Userland/Applications/PixelPaint/LevelsDialog.cpp

```diff
@@ -94,15 +88,34 @@ void LevelsDialog::generate_new_image()
     Color current_pixel_color;
     Color new_pixel_color;
     Gfx::StorageFormat storage_format = Gfx::determine_storage_format(m_editor->active_layer()->content_bitmap().format());
+    auto apply_only_on_mask = m_editor->active_layer()->mask_type() == Layer::MaskType::AlphaMask;
 
     for (int x = 0; x < m_reference_bitmap->width(); x++) {
         for (int y = 0; y < m_reference_bitmap->height(); y++) {
             current_pixel_color = m_reference_bitmap->get_pixel(x, y);
 
-            new_pixel_color.set_alpha(current_pixel_color.alpha());
-            new_pixel_color.set_red(m_precomputed_color_correction[current_pixel_color.red()]);
-            new_pixel_color.set_green(m_precomputed_color_correction[current_pixel_color.green()]);
-            new_pixel_color.set_blue(m_precomputed_color_correction[current_pixel_color.blue()]);
+            if (apply_only_on_mask) {
```

Comment:
That makes totally sense, i'll try to create a function on the layer so that pixel manipulation with respect to a masks can be done on a general level.